unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Catalog/QuickBarController.cs

```diff
@@ -0,0 +1,118 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+public class QuickBarController : MonoBehaviour
+{
+    public RawImage[] shortcutsImgs;
```

Comment:
This array points me to something missing in the architecture. I believe an entity `QuickBarItemSlot` is asking for existance. 

```
public class QuickBarItemSlot : Monobehaviour
{
    [SerializeField] private RawImage image;
    private SceneObject sceneObject;

    public void SetSceneObject(SceneObject object);
    public void SetEmpty();
}
```

Therefore `QuickBarController` would have an array of these `QuickBarItemSlot` and they would be self sufficient and more testable.

Reply:
Obviously this array and such would go into a `View` class. The Controller would just need the `SceneObjects`

Reply:
The problem is that the SceneObject doesn't contain the texture, but I will create it with the texture as parameter

Reply:
Sure but the `SceneObject` can build the thumbnail URL right? Therefore we can just use AssetPromise within this new `QuickBarItemSlot` to get the texture. You dont have to worry about said texture being downloaded multiple times because the AssetPromise system will take care of that and cache stuff.

Reply:
I already have the texture when the scene object is added, so I think it is more performance-wise to just assign the reference instead of asking the AssetPromise if it has the reference (It will always have the texture because if there is no texture you can't use the scene object, however, it is not the most beautiful from an architecture point of view

Reply:
You dont really ask the AssetPromise, just request the asset by an id (in this case the URL) and the system will take care of everything for you. The cool thing about the AssetPromise (and the AssetLibrary built behind) is that you dont really care whether the asset is cached, has (or is in the process) to be downloaded. Also requesting a cached asset is O(0) since it's in a dictionary.

I agree passing the Texture is more perfomant, but I think the gain is marginally and the benefits outscale it.
The `QuickBarItemSlot` would be more independent and self-sufficent, it would make it more testable and the code would be easier to read/extend since you would decrease the amount of params to track down when following the architecture. 

In any case, this is something we can tackle down later, it is good as it is right now.